### ISSUE TITLE
Port select, poll and epoll reactors to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6068]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6068|
|Type|enhancement|
|Created|2012-10-05 19:38:58Z|
|Branch|https://github.com/twisted/twisted/tree/reactors-py3-6068|

We've got enough test coverage at this point in trunk, or branches, that porting select, poll and epoll seems reasonable.

<details><summary>Searchable metadata</summary>

```
trac-id__6068 6068
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_reactors_py3_6068 branches/reactors-py3-6068
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349465938000000 1349465938000000
changetime__1350314666000000 1350314666000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35992]) Branching to 'reactors-py3-6068'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/reactors-py3-6068

Nothing very exciting here.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Urrr really?  How is that possible?  I guess even if the modules are "ported" that doesn't mean the reactor-related tests in `twisted.internet.test` are passing.  So... okay?  I guess?  Clearly I'm slightly confused by this, but it doesn't seem _wrong_...  Go ahead and merge, if you know where you're going next (maybe write down where you're going next though).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

These modules only really provide `IReactorFDSet`, so it seems reasonable to merge this once `test_udp` and `test_tcp` are passing.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [36155]) Merge reactors-py3-6068.

Author: itamar
Review: exarkun
Fixes: [#6068](https://github.com/twisted/twisted/issues/6068)

select, poll and epoll reactors are deemed to be sufficiently working on Python 3, given passing TCP, UDP and fdset tests.
